### ISSUE TITLE
Inconvenient emoji ascii replacement

### ISSUE THREAD

**Issue by felpsio** [tommoor/react-emoji-render#96]
The regex that identifies ascii emojis and replaces them by the actual emojis should check for whitespace before and after the match. Or it may add inconvenient emojis in some cases. Example:

`"chrome://settings/content/notifications"` is translated to `"chrome😕/settings/content/notifications"`

Suggestion:

only apply the ascii conversion when:
- the emoji is between "start of line" and "end of line"
- the emoji is between whitespace

**Comment by felpsio** [comment]
To solve my user case I updated the asciiRegex to:

```
(?: |^)(${edgeCases})?(${names}|:)([${allowedAliasCharacters}]*:)?(?: |$)
```

I basically added `(?: |^)` and `(?: |$)`

This forces the regex to make a emoji substitution only when the ascii match is separated by a whitespace or start/end of line

**Comment by enzoferey** [comment]
Hey @felpsio, `chrome://` URLs are indeed edge cases. I prepared the regex to support those kind of issues via the `edgeCases` variable that be found here:

https://github.com/tommoor/react-emoji-render/blob/e959d6d0f6fea73d0392c6b5df5fb336cada551b/src/aliasRegex.js#L7-L15

In the same way we have `startOfURL`, we could have `startOfChromeURL` and cover that.

**Comment by felpsio** [comment]
hey @enzoferey, but this can help on many other cases.

For example, instead of chrome: `safari, brave, firefox, file, base64 etc.`

So it's probably safer require a whitespace or start/end of line to avoid all these cases.

Still going to work on these cases: `:)`, ` :)`, ` :) `. But it won't work on `<any string without whitespace>:)<any string without whitespace>`

**Comment by enzoferey** [comment]
Right, the design decisions for this library was to support emojis without required whitespaces, hence going for this solution. I'm happy to add as many edge cases as needed.

Note that you can always escape certain parts of a string, transform the rest of the string with this library, and then concatenate back.

**Comment by felpsio** [comment]
Thanks @enzoferey.

You're right. My use case may not suit the goal of this lib. So for now, I'll probably keep my forked version and bring the updates from the main version manually as they update.

But I suggest then adding these edge cases: `brave://`, `chrome://` and `file://`.

Feel free to close my issue then :).

If somebody else needs to add `whitespaces` as a requirement, you can use my adapted version: https://github.com/felpsio/react-emoji-render or just fork and apply the same regex change

**Comment by enzoferey** [comment]
As you prefer @felpsio, happy to accept PRs for such edge cases 👍🏻

**Comment by nicolasvidelac** [comment]
Hi @felpsio, I encountered an issue really close to this one so i didn't feel like opening a new issue was necessary. 
If i have the text: "https://somexample.com/tommoor/react-emoji-render/issues/96", it will not add an emoji to it. 

But, if i have "Https" instead of "https", it will, so "Https://somexample.com/issues/96" becomes "Https😕/somexample.com/issues/96".

So, the case should not matter when checking if it's an exception or not.

**Comment by felpsio** [comment]
@nicolasvidelac maybe for your case you can just transform the string in lower case. `Https` is not an usual way for URLs, so change the regex to accept the upper case H may cause some undesired side effects or other edge cases.
